src/compiler/checker.ts

```diff
@@ -24005,6 +23995,17 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
         if (inInferTypeForHomomorphicMappedType) {

```

Comment:
tangent q: is there any reason why this is a "global" flag and not something based on source/target identities? 🤔